plerkle/src/macros.rs

```diff
@@ -0,0 +1,202 @@
+use anchor_client::anchor_lang;
+use std::borrow::{Borrow, BorrowMut};
+use std::collections::HashMap;
+use std::{
+    fmt::{Debug, Formatter},
+    ops::Index,
+    rc::Rc,
+    cell::RefCell,
+};
+use std::result::Iter;
+
+extern crate redis;
+
+use crate::error::PlerkleError;
+use anchor_lang::Event;
+use redis::streams::StreamMaxlen;
+use redis::Commands;
+use redis::{Client, Connection, RedisResult};
+use regex::Regex;
+use solana_geyser_plugin_interface::geyser_plugin_interface::{GeyserPluginError, ReplicaAccountInfoVersions, ReplicaBlockInfoVersions, ReplicaTransactionInfo, ReplicaTransactionInfoVersions, Result, SlotStatus};
+use std::str::FromStr;
+use solana_sdk::instruction::CompiledInstruction;
+use solana_sdk::message::AccountKeys;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::{keccak, pubkeys};
+use solana_sdk::transaction::Transaction;
+use anchor_client::anchor_lang::AnchorDeserialize;
+use hex;
+use {
+    log::*,
+    solana_geyser_plugin_interface::geyser_plugin_interface::GeyserPlugin,
+    std::{fs::File, io::Read},
+    thiserror::Error,
+};
+use gummyroll_crud::InstructionName;
+use crate::programs::gummy_roll::handle_change_log_event;
+
+#[macro_use]
+pub mod macros {
+    use super::*;
+
+    macro_rules! define_redis_plugin {
```

Comment:
I think using a trait is preferable to setting up a macro here. Partially a matter of personal preference, but using traits and implementing them is probably easier for the average Rust dev to integrate with